File: 백준/Silver/27376. 참살이길/참살이길.py
def cw(time, period, start):
    if time<start:
        return start - time
    now = time - start
    sw = (now//period)%2
    return sw * (period - now % period)

temp = list(map(int,input().split()))
n = temp[0]
k = temp[1]
cross = []
for i in range(k):
    x = list(map(int,input().split()))
    cross.append(x)
cross.sort(key=lambda x:x[0])
mytime = 0
mypos = 0

for x,t,s in cross:
    mytime += x-mypos
    mytime += cw(mytime,t,s)
    mypos = x
mytime = mytime + n-mypos
print(mytime)

File: 백준/Silver/27376. 참살이길/참살이길.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const [n, k] = input[0].split(" ").map(Number)
const lights = input.slice(1).map((row) => row.split(" ").map(Number))
lights.sort((a, b) => a[0] - b[0])

let dist = 0
let time = 0

const cal = (time, period, start) => {
  if (time < start) {
    return start - time
  }
  const rest = time - start

  const flag = Math.floor(rest / period) % 2 // 홀수면 빨 // 짝수면 초

  if (flag === 1) {
    // 다음 초록으로 변할때까지의 시간을 return
    return period - (rest % period)
  } else {
    return 0
  }
}

for (let [pos, period, start] of lights) {
  time += pos - dist
  time += cal(time, period, start)
  dist = pos
}
time += n - dist
console.log(time)